tests/http/test_proxyagent.py

```diff
@@ -137,48 +140,46 @@ def test_http_request(self):
         body = self.successResultOf(treq.content(resp))
         self.assertEqual(body, b"result")
 
+    def test_http_request(self):
+        agent = ProxyAgent(self.reactor)
+        self._test_request_no_proxy(agent, b"http", b"test.com", b"")
+
     def test_https_request(self):
         agent = ProxyAgent(self.reactor, contextFactory=get_test_https_policy())
+        self._test_request_no_proxy(agent, b"https", b"test.com", b"abc")
 
-        self.reactor.lookups["test.com"] = "1.2.3.4"
-        d = agent.request(b"GET", b"https://test.com/abc")
-
-        # there should be a pending TCP connection
-        clients = self.reactor.tcpClients
-        self.assertEqual(len(clients), 1)
-        (host, port, client_factory, _timeout, _bindAddress) = clients[0]
-        self.assertEqual(host, "1.2.3.4")
-        self.assertEqual(port, 443)
-
-        # make a test server, and wire up the client
-        http_server = self._make_connection(
-            client_factory,
-            _get_test_protocol_factory(),
-            ssl=True,
-            expected_sni=b"test.com",
+    def test_http_request_via_no_proxy(self):
+        agent = ProxyAgent(
+            self.reactor, http_proxy=b"proxy.com:8888", no_proxy=b"test.com,unused.com"
```

Comment:
Could you update the tests to instead inject proxy information into the OS environment, in order to test that Synapse reads it correctly? This should be as easy as adding a decorator similar to:

```py
@mock.patch.dict(os.environ, {"http_proxy": "proxy.com:8888", "no_proxy": "test.com,unused.com"})
```

to the header of test functions.